libs/providers/flagd/src/lib/flagd-provider.ts

```diff
@@ -3,45 +3,38 @@ import {
   Provider,
   ResolutionDetails,
 } from '@openfeature/nodejs-sdk';
-import { Service } from './service/Service';
-import { HTTPService } from './service/http/service';
+import { Service } from './service/service';
 import { GRPCService } from './service/grpc/service';
+import { Protocol } from './service/grpc/protocol';
 
 export interface FlagdProviderOptions {
   service?: 'grpc' | 'http';
   host?: string;
   port?: number;
-  protocol?: 'http' | 'https';
+  protocol?: Protocol;
 }
 
 export class FlagdProvider implements Provider {
   metadata = {
-    name: 'flagD Provider',
+    name: 'flagd Provider',
   };
 
   private readonly service: Service;
 
-  constructor(options?: FlagdProviderOptions) {
-    const { service, host, port, protocol }: FlagdProviderOptions = {
+  constructor(options?: FlagdProviderOptions, service?: GRPCService) {
```

Comment:
This is a non-breaking changes that allows the injection of a GRPCService instance. Few will probably use this, but it makes testing much easier.